### PR TITLE
Enable `whyNoLint` and `ifElseChain` `gocritic` linters

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
This enables the rest of the missing gocritic linters and fixes their reports.

#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
@kubernetes/release-engineering PTAL

File: cmd/krel/cmd/changelog_data_test.go [hunk 1]
```diff
@@ -870,7 +870,7 @@ const minorReleaseExpectedContent = `# Changelog since v1.20.0
 
 PSP as an admission controller resource is being deprecated. Deployed PodSecurityPolicy's will keep working until version 1.25, their target removal from the codebase. A new feature, with a working title of "PSP replacement policy", is being developed in [KEP-2579](https://features.k8s.io/2579). To learn more, read [PodSecurityPolicy Deprecation: Past, Present, and Future](https://blog.k8s.io/2021/04/06/podsecuritypolicy-deprecation-past-present-and-future/).`
 
-// nolint: misspell
+// nolint:misspell // there is a spelling mistage in the notes
```
Comment (xmudrii) [inline cmd/krel/cmd/changelog_data_test.go:873]:
```suggestion
// nolint:misspell // there is a spelling mistake in the notes
```

Unless it's an intentional typo... :smile:

**Review by xmudrii — APPROVED** [review]

File: cmd/krel/cmd/changelog_data_test.go (hunk 1 above)
Comment (saschagrunert) [reply cmd/krel/cmd/changelog_data_test.go]: Great catch! Fixed as suggested :+1:

**Pull request merged**